foreign_cc/nmake.bzl

```diff
@@ -0,0 +1,59 @@
+"""Helper rules and macros for building projects using the MSVC nmake build tool"""
+
+load(":configure.bzl", "configure_make")
+load("//foreign_cc:providers.bzl", "ForeignCcDepsInfo")
+
+def _extra_toolchains_transition_impl(settings, attrs):
+    return {"//command_line_option:extra_toolchains": settings["//command_line_option:extra_toolchains"] + attrs.extra_toolchains}
+
+_extra_toolchains_transition = transition(
+    implementation = _extra_toolchains_transition_impl,
+    inputs = ["//command_line_option:extra_toolchains"],
+    outputs = ["//command_line_option:extra_toolchains"],
+)
+
+def _extra_toolchains_transitioned_foreign_cc_target_impl(ctx):
+    target = ctx.attr.target[0]
+
+    # Return the providers from the transitioned foreign_cc target
+    return [
+        target[DefaultInfo],
+        target[CcInfo],
+        target[ForeignCcDepsInfo],
+        target[OutputGroupInfo],
+    ]
+
+extra_toolchains_transitioned_foreign_cc_target = rule(
+    doc = "A rule for adding extra toolchains to consider when building the given target",
+    implementation = _extra_toolchains_transitioned_foreign_cc_target_impl,
+    attrs = {
+        "extra_toolchains": attr.string_list(
+            doc = "Additional toolchains to consider",
+            mandatory = True,
+        ),
+        "target": attr.label(
+            doc = "The target to build after considering the extra toolchains",
+            cfg = _extra_toolchains_transition,
+            providers = [ForeignCcDepsInfo],
+            mandatory = True,
+        ),
+        "_allowlist_function_transition": attr.label(
+            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
+        ),
+    },
+    incompatible_use_toolchain_transition = True,
+)
+
+def configure_nmake(name, **kwargs):
+    """ Wrapper macro around configure_make to force usage of the preinstalled nmake toolchain """
+    configure_make_target_name = name + "_nmake"
+
+    configure_make(
+        name = configure_make_target_name,
+        **kwargs

```

Comment:
Don't you need some special handling of deps to transition them back to the original `make_toolchain`? Otherwise transitive deps will try to use `nmake` in place of `make`?

Reply:
You are right, as I had mistakenly implemented an "outgoing edge" transition. I have now modified it to be an "incoming edge" transition, which according to the [docs](https://docs.bazel.build/versions/main/skylark/config.html#attaching-transitions) should apply to the provided target only